src/tests/ftest/control/dmg_telemetry_net.py

```diff
@@ -0,0 +1,166 @@
+from telemetry_test_base import TestWithTelemetry
+from general_utils import DaosTestError, pcmd
+from command_utils import CommandFailure
+
+# pylint: disable=too-few-public-methods,too-many-ancestors
+# pylint: disable=attribute-defined-outside-init
+class TestWithTelemetryNet(MdtestBase, TestWithTelemetry):
+    """Rebuild test cases featuring mdtest.
+
+    This class contains tests for pool rebuild that feature I/O going on
+    during the rebuild using IOR.
+
+    :avocado: recursive
+    """
+
+    def pre_tear_down(self):
+        """Tear down any test-specific steps prior to running tearDown().
+
+        Returns:
+            list: a list of error strings to report after all tear down
+            steps have been attempted
+
+        """
+        # kill any hanging mdtest processes
+        ret = []
+
+        # DEBUGGING
+        pcmd(self.hostlist_servers, "ps aux")
+
+        cmd = "pkill -9 mdtest"
+        for retry in range(0, 5):
+            ret_codes = pcmd(self.hostlist_servers, cmd)
+            if len(ret_codes) > 1 or 0 not in ret_codes:
+                self.log.info("Retry #{}: Not all '{}' commands in pcmd "
+                              "returned 0.".format(retry, cmd))
+
+        # Ignore errors cleaning up mdtest
+        return []
+
+    def tearDown(self):
+        try:
+            super().tearDown()
+        except (CommandFailure, DaosTestError) as excep:
+            self.log.info("{}: tearDown threw an exception "
+                          "because a rank was (intentionally) "
+                          "stopped midway through a run of mdtest."
+                          .format(repr(excep)))
+            pass
+
+    def test_net_telemetry(self):
+        """Jira ID: DAOS-9020.
+
+        Test Description: Verify engine net telemetry metrics.
+
+        Use Cases:
+          * Create pool and container.
+          * Use mdtest to create 120K files with 3-way replication.
+          * Stop one server.
+          * Ensure engine_net_ofi_sockets_req_timeout is greater than 0.
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,large
+        :avocado: tags=control,telemetry,net
+        :avocado: tags=test_with_telemetry_net,test_net_telemetry
+        """
+
+        # Get the req_timeout metric name, which depends on the libfabric
+        # provider name
+        all_metrics = self.telemetry.get_all_server_metrics_names(
+            self.server_managers[0])
+        p = r"engine_net_\w+_req_timeout"
+        metrics = list(filter(lambda x: search(p, x), all_metrics))
+
+        # Initialize req_timeouts based on what's already populated in the
+        # telemtry system, which may or may not be zero.
+        req_timeouts_start = 0
+
+        data = {}
+        try:
+            data = self.telemetry.get_metrics(metrics[0])
+        except (CommandFailure, DaosTestError) as excep:
+            self.log.info("self.telemetry.get_metrics failed on at least one "
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions